src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -27,74 +51,585 @@ const ActorInit En_Dns_InitVars = {
+            this->unk_2CE = 0;
+            this->unk_2C6 |= 0x10;
+        }
+        this->actionFunc = func_8092D1B8;
+    } else {
+        Math_ApproachS(&this->actor.shape.rot.y, sp2E, 3, 0x2AA8);
+    }
+}
+
+void func_8092D5E8(EnDns* this, GlobalContext* globalCtx) {
+    static s32 D_8092DE0C[] = { 0, 0, 4, 6 };
+    u32 temp_v0;
+    u32 temp_v1;
+
+    if (func_800EE29C(globalCtx, this->unk_2C8)) {
+        temp_v0 = func_800EE200(globalCtx, this->unk_2C8);
+        temp_v1 = globalCtx->csCtx.npcActions[temp_v0]->unk0;
+        if (this->unk_1D8 != (u8)temp_v1) {
+            func_8092C63C(this, D_8092DE0C[temp_v1]);
+            this->unk_1D8 = temp_v1;
+        }
+
+        if (((this->unk_2F8 == 4) || (this->unk_2F8 == 6)) &&
+            func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+            func_8092C63C(this, this->unk_2F8 + 1);
+        }
+
+        func_800EDF24(&this->actor, globalCtx, temp_v0);
+    }
+}
+
+void EnDns_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDns* this = THIS;
+
+    if (!func_8092D068(this)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 18.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06002DD8, NULL, this->jointTable, this->morphTable, 13);
+    this->unk_2F8 = -1;
+    func_8092C63C(this, 2);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.targetMode = 0;
+    this->actor.gravity = -0.8f;
+    this->unk_2D2 = 0;
+    this->unk_2C6 = 0;
+    func_8013AED4(&this->unk_2C6, 3, 7);
+    this->unk_2C6 |= (0x40 | 0x10);
+    this->unk_2C6 |= 0x200;
+    if (gSaveContext.weekEventReg[9] & 0x80) {
+        this->unk_2FC = 1;
+    } else {
+        this->unk_2FC = 0;
+    }
+    this->actionFunc = func_8092D1B8;
+    gSaveContext.eventInf[1] &= (u8)~0x20;
+    gSaveContext.eventInf[1] &= (u8)~0x40;
+}
+
+void EnDns_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDns* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnDns_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDns* this = THIS;
+
+    if (!func_8092CAD0(this, globalCtx) && func_8092CB98(this, globalCtx)) {
+        func_8092D5E8(this, globalCtx);
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+        func_8092C5C0(this);
+        return;
+    }
+
+    this->actionFunc(this, globalCtx);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_8092C934(this);
+    func_8092C86C(this, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 12.0f, 0.0f, 4);
+    func_8013C964(&this->actor, globalCtx, 80.0f, 40.0f, 0, this->unk_2C6 & 7);
+    Actor_SetHeight(&this->actor, 34.0f);
+    func_8092C6FC(this, globalCtx);
+    func_8092C5C0(this);
```

Comment:
```suggestion
    if (!func_8092CAD0(this, globalCtx) && func_8092CB98(this, globalCtx)) {
        func_8092D5E8(this, globalCtx);
        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
        func_8092C5C0(this);
    } else {
        this->actionFunc(this, globalCtx);
        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
        func_8092C934(this);
        func_8092C86C(this, globalCtx);
        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 12.0f, 0.0f, 4);
        func_8013C964(&this->actor, globalCtx, 80.0f, 40.0f, 0, this->unk_2C6 & 7);
        Actor_SetHeight(&this->actor, 34.0f);
        func_8092C6FC(this, globalCtx);
        func_8092C5C0(this);
    }
```
Assuming it matches, an else should be able to get rid of the return

Reply:
I'm personally fine with an early return.